hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/dynamicconfig/WanReplicationConfigTransformer.java

```diff
@@ -0,0 +1,275 @@
+        return new WanBatchPublisherConfigHolder(
+                config.getPublisherId(),
+                config.getClassName(),
+                serializationService.toData(config.getImplementation()),
+                toHolderProperties(config.getProperties()),
+                config.getClusterName(),
+                config.isSnapshotEnabled(),
+                config.getInitialPublisherState().getId(),
+                config.getQueueCapacity(),
+                config.getBatchSize(),
+                config.getBatchMaxDelayMillis(),
+                config.getResponseTimeoutMillis(),
+                config.getQueueFullBehavior().getId(),
+                config.getAcknowledgeType().getId(),
+                config.getDiscoveryPeriodSeconds(),
+                config.getMaxTargetEndpoints(),
+                config.getMaxConcurrentInvocations(),
+                config.isUseEndpointPrivateAddress(),
+                config.getIdleMinParkNs(),
+                config.getIdleMaxParkNs(),
+                config.getTargetEndpoints(),
+                config.getAwsConfig(),
+                config.getGcpConfig(),
+                config.getAzureConfig(),
+                config.getKubernetesConfig(),
+                config.getEurekaConfig(),
+                serializationService.toData(config.getDiscoveryConfig()),
+                config.getSyncConfig().getConsistencyCheckStrategy().getId(),
+                config.getEndpoint()
+        );
+    }
+
+    @Nonnull
+    Map<String, Comparable> toProperties(@Nullable Map<String, Data> properties) {
+        if (properties == null) {
+            return Collections.emptyMap();
+        }
+
+        Map<String, Comparable> requiredProperties = new HashMap<>();
+        for (Map.Entry<String, Data> e : properties.entrySet()) {
+            Comparable value = serializationService.toObject(e.getValue());
+            // we don't check the [value] if it's null
+            requiredProperties.put(e.getKey(), value);
+        }
+        return requiredProperties;
+    }
+
+    @Nullable
+    WanConsumerConfig toConfig(@Nullable WanConsumerConfigHolder holder) {
+        if (holder == null) {
+            return null;
+        }
+
+        WanConsumerConfig config = new WanConsumerConfig();
+        config.setProperties(toProperties(holder.getProperties()));
+        config.setPersistWanReplicatedData(holder.isPersistWanReplicatedData());
+
+        // this is all duplicated rubbish
+        // note. class name and implementation are mutually exclusive -- className seems the safer of the two
+        boolean classNameProvided = holder.getClassName() != null;
+        if (classNameProvided) {
+            config.setClassName(holder.getClassName());
+        }
+        if (!classNameProvided && holder.getImplementation() != null) {
+            WanConsumer wanConsumer = serializationService.toObject(holder.getImplementation());
+            config.setImplementation(wanConsumer);
+        }
```

Comment:
@vbekiaris `className` and `implementation` are mutually exclusive in all the configs. I would think that `className` is safer to use due to use of serialization requirements on the actual implementation, e.g. in this case the `WanConsumer`.

What are your thoughts on this? It happens in a couple of configs -- same semantics (and logic on my part for now).